src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala

```diff
@@ -618,4 +617,31 @@ final case class EServicesApiServiceImpl(
       }
     }
   }
+
+  override def deleteEService(
+    eServiceId: String
+  )(implicit contexts: Seq[(String, String)], toEntityMarshallerProblem: ToEntityMarshaller[Problem]): Route = {
+
+    def deleteEserviceIfEmpty(eService: CatalogProcessEService, isDraftDeleted: Boolean = true): Future[Unit] =
+      if (eService.descriptors.isEmpty) {
+        catalogProcessService.deleteEService(eService.id)
+        Future.successful(()) // Adding this in case the eService is already empty in order to not repeat the delete
+        // Eservice at the end of the for
+      } else if (isDraftDeleted) Future.successful(())
+      else Future.failed(EServiceCannotBeDeleted(eService.id.toString))
```

Comment:
is it confirmed by the frontend that in this case we should return an error?

Reply:
Yes, in case there's nothing to delete they expect an Error instead of a "silent success" that we suggested